lib/StringClient.js

```diff
@@ -18,14 +18,27 @@ var HttpClient = require('./HttpClient');
 function StringClient(options) {
     assert.object(options, 'options');
     assert.optionalObject(options.gzip, 'options.gzip');
+    assert.optionalObject(options.contentMd5, 'options.contentMd5');
+
+    if (options.contentMd5) {
+        assert.optionalArrayOfString(options.contentMd5.encodings,
```

Comment:
Do we want to validate that the provided encodings are one of 'utf8' and 'binary', or do we want to leave that open so that we can easily extend what encodings are supported?

If we specify a non-existing encoding like `'foo'`, what happens? Does the crypto module throw an error?

Reply:
Yes - I've now added a validation check for the contentMd5.encoding array, which checks Buffer.isEncoding.

Reply:
Awesome, thank you!